crates/red_knot/tests/cli.rs

```diff
@@ -0,0 +1,44 @@
+use anyhow::Context;
+use insta_cmd::{assert_cmd_snapshot, get_cargo_bin};
+use std::process::Command;
+use tempfile::TempDir;
+
+/// Specifying an option on the CLI should take precedence over the same setting in the
+/// project's configuration.
+#[test]
+fn test_config_override() -> anyhow::Result<()> {
+    let tempdir = TempDir::new()?;
+
+    std::fs::write(
+        tempdir.path().join("pyproject.toml"),
+        r#"
+[tool.knot.environment]
+python-version = "3.11"
+"#,
+    )
+    .context("Failed to write settings")?;
+
+    std::fs::write(
+        tempdir.path().join("test.py"),
+        r#"
+import sys
+
+print(sys.last_exc)
+"#,
+    )
+    .context("Failed to write test.py")?;
+
+    assert_cmd_snapshot!(knot().arg("--project").arg(tempdir.path()).arg("--python-version").arg("3.12"), @r"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    ");
```

Comment:
Can we also assert that the same command fails without the `--python-version 3.12` option?